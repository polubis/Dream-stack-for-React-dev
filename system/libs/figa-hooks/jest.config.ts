/* eslint-disable */
export default {
  displayName: 'figa-hooks',
  preset: '../../jest.preset.js',
  transform: {
    '^(?!.*\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/figa-hooks',
};
