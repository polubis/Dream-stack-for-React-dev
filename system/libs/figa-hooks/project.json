{
  "name": "figa-hooks",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/figa-hooks/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/figa-hooks/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nrwl/vite:build",
      "outputs": ["{options.outputPath}"],
      "defaultConfiguration": "production",
      "options": {
        "outputPath": "dist/libs/figa-hooks"
      },
      "configurations": {
        "development": {
          "mode": "development"
        },
        "production": {
          "mode": "production"
        }
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/figa-hooks/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  }
}
