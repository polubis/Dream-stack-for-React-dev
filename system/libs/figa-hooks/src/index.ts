export * from './lib/use-portal';
export * from './lib/use-toggle';
export * from './lib/use-intersection-observer';
export * from './lib/use-click-outside';
export * from './lib/use-stepper';
export * from './lib/use-scroll-y';
export * from './lib/use-client-effect';
export * from './lib/use-client-layout-effect';
export * from './lib/use-element-size';
export * from './lib/use-clipboard';
export * from './lib/use-popover';
