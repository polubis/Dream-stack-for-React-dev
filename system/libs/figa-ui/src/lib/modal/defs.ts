import type { BoxProps } from '../box';

interface ModalProps extends BoxProps {
  onClose?: () => void;
}

export type { ModalProps };
