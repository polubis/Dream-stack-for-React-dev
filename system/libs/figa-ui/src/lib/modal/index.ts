export * from './modal';
export * from './defs';
