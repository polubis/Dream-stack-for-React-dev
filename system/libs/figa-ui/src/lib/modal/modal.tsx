import type { ModalProps } from './defs';

import styled from 'styled-components';
import { usePortal } from '@system/figa-hooks';
import { tokens } from '../theme-provider';

const Backdrop = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: ${(props) => props.theme.modal.backdrop};
  z-index: ${tokens.z[450]};
`;

const Wrapper = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  min-width: 280px;
  min-height: 280px;
  max-height: 96vh;
  padding: ${tokens.padding[250]};
  transform: translate(-50%, -50%);
  background: ${(props) => props.theme.modal.background};
  border-radius: ${tokens.radius[50]};
  border: 1px solid ${(props) => props.theme.modal.border};
  z-index: ${tokens.z[500]};
  overflow-y: auto;
`;

export const Modal = ({ children, onClose }: ModalProps) => {
  const { render } = usePortal();

  return render(
    <>
      <Backdrop onClick={onClose} />
      <Wrapper>{children}</Wrapper>
    </>
  );
};
