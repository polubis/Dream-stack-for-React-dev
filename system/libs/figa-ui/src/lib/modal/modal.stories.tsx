import type { Story, Meta } from '@storybook/react';
import type { ModalProps } from './defs';

import { Modal } from './modal';
import { useState } from 'react';

export default {
  component: Modal,
  title: 'Modal',
} as Meta;

const Template: Story<ModalProps> = (args) => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <button onClick={() => setOpen(!open)}>Click me to open :)</button>
      {open && <Modal {...args} onClose={() => setOpen(false)} />}
    </>
  );
};

export const Basic = Template.bind({});
Basic.args = {
  children: (
    <div style={{ width: '50px', height: '50px' }}>Content inside modal</div>
  ),
};
