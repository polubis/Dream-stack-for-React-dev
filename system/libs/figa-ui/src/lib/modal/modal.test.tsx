import { render } from '@testing-library/react';
import { Modal } from './modal';
import { type ReactNode } from 'react';

import { usePortal } from '@system/figa-hooks';

jest.mock('@system/figa-hooks');

describe('Modal can be used when: ', () => {
  beforeAll(() => {
    (usePortal as jest.Mock).mockImplementation(() => ({
      render: (children: ReactNode) => children,
    }));
  });

  it('[FRAGILE] assigns classes', () => {
    const { asFragment } = render(<Modal className="my-class" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
