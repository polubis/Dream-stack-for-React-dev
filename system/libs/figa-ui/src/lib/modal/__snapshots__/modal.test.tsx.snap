// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  />
  <div
    class="box modal my-class column empty"
  />
</DocumentFragment>
`;
